### PR TITLE
chore: simplifying linting for cli/types

### LINKED ISSUE #26394: Using a community/custom component framework definition causes a TypeScript error
### Current behavior

When you use an unofficial framework in your cypress.config.ts, TypeScript complains.

### Desired behavior

You should be allowed to use framework definitions that fit the format `cypress-ct-[name]` or `@[scope]/cypress-ct-[name]`.

### Test code to reproduce

cypress.config.ts
```ts
import { defineConfig } from 'cypress';
import { defineConfig as defineViteConfig } from 'vite';

export default defineConfig({
  component: {
    devServer: {
      framework: 'cypress-ct-lit-element', // Error on this line
      bundler: 'vite'
    }
  }
});

```

### Cypress Version

12.9.0

### Node version

16.16.0

### Operating System

MacOS 13.2.1

### Debug Logs

_No response_

### Other

_No response_

### PR DESCRIPTION
**Pull request by lmiller1990**
- Closes https://github.com/cypress-io/cypress/issues/26394

### Additional details

The change in https://github.com/cypress-io/cypress/issues/26394 requires TS 4.1. We tried to just set the minimum TS version to 4.1, but `dtslint` was erroring out for unknown reasons when changing this. I am completely unable to run `dtslint` locally.

This prompted us to try updating `dtslint` to the latest version, to find it was deprecated in favor of a new repo, https://www.npmjs.com/package/@definitelytyped/dtslint. This is a pre 1.0 package and is quite buggy. It also uses `tslint` under the hood, which is also deprecated.

To reduce complexity and avoid using deprecated and/or pre 1.0 libraries, I moved our type linting and checking in the `cli` package to just use `tsc`, which is a lot easier to run and debug locally, and doesn't require any external dependencies.

In summary:

1. remove deprecated dtslint and tslint in favor of plain old `tsc`
2. apply repo eslint rules to `cl/types`

I did not change the minimum TS version yet, but I think we should make it >= 4.1 in Cypress 13.

### Steps to test

Run `yarn types` in `cli` locally.

### PR Tasks

- [x] Have tests been added/updated?
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [ ] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

File: .circleci/workflows.yml [hunk 1]
```diff
@@ -1489,11 +1489,7 @@ jobs:
           working_directory: cli/types
       - run:
           name: "Lint types 🧹"
-          command: yarn workspace cypress dtslint
-  # todo(lachlan): do we need this? yarn check-ts does something very similar
-  #     - run:
-  #         name: "TypeScript check 🧩"
-  #         command: yarn type-check --ignore-progress
```
Comment (lmiller1990) [inline .circleci/workflows.yml:1496]: Output: https://app.circleci.com/pipelines/github/cypress-io/cypress/52006/workflows/846acd8d-3f61-4476-a31f-3d9fe6fa91bc/jobs/2146111

File: cli/.eslintignore [hunk 2]
```diff
@@ -1,14 +1,21 @@
...
 /react
 /react18
 /vue
 /vue2
 /svelte
-/mount-utils
\ No newline at end of file
+/mount-utils
+
+/types/bluebird/**/*
```
Comment (lmiller1990) [inline cli/.eslintignore:13]: We only lint and format our own types, not the ones we copy-paste from node_modules. These are `.gitignored`.

File: cli/types/cypress.d.ts [hunk 3]
```diff
@@ -3466,13 +3466,15 @@ declare namespace Cypress {
   type ConfigHandler<T> = T
   | (() => T | Promise<T>)
 
+  type CustomFramework = `cypress-ct-${string}` | `@${string}/cypress-ct-${string}`
```
Comment (lmiller1990) [inline cli/types/cypress.d.ts]: Since we now use eslint (same as rest of repo) instead of the deprecated `tslint`, the types now match the rest of the formatting rules in our repo.

**Comment by lmiller1990** [comment]
I will come back to this one day, but not right now.
